pkgs/development/compilers/openjdk/openjfx/16.nix

```diff
@@ -69,47 +69,73 @@ let
+    patches = [
+      # libglass.so tries to load GTK to see it is available, so we must add GTK
+      # to the RUNPATH of the libglass.so.
+      ./add_gtk_to_libglass.patch
+      # libstreamer-lite.so loads FFmpeg at runtime, so we must add FFmpeg to the
+      # RUNPATH of the libstreamer-lite.so.
+      ./add_ffmpeg_to_libgstreamer-lite.patch
+    ];
+
+    gradleProperties = ''
+      COMPILE_MEDIA = true
+      COMPILE_WEBKIT = true
+    '';
 
-  gradleProperties = ''
-    COMPILE_MEDIA = true
-    COMPILE_WEBKIT = true
-  '';
+    preBuild = ''
+      swtJar="$(find ${deps} -name org.eclipse.swt\*.jar)"
+      substituteInPlace build.gradle \
+        --replace 'mavenCentral()' 'mavenLocal(); maven { url uri("${deps}") }' \
+        --replace 'name: SWT_FILE_NAME' "files('$swtJar')"
+    '';
 
-  preBuild = ''
-    swtJar="$(find ${deps} -name org.eclipse.swt\*.jar)"
-    substituteInPlace build.gradle \
-      --replace 'mavenCentral()' 'mavenLocal(); maven { url uri("${deps}") }' \
-      --replace 'name: SWT_FILE_NAME' "files('$swtJar')"
-  '';
+    stripDebugList = [ "." ];
 
-  installPhase = ''
-    cp -r build/modular-sdk $out
-  '';
+    postFixup = ''
+      # Remove references to bootstrap.
+      find "$out" -name \*.so | while read lib; do
+        new_refs="$(patchelf --print-rpath "$lib" | sed -E 's,:?${lib.escape ["+"] openjdk11_headless.outPath}[^:]*,,')"
+        patchelf --set-rpath "$new_refs" "$lib"
+      done
+    '';
 
-  # glib-2.62 deprecations
-  NIX_CFLAGS_COMPILE = "-DGLIB_DISABLE_DEPRECATION_WARNINGS";
+    disallowedReferences = [ openjdk11_headless ];
+  } // args);
 
-  stripDebugList = [ "." ];
+in {
+  modular-sdk = common {
+    pname = "openjfx-modular-sdk";
 
-  postFixup = ''
-    # Remove references to bootstrap.
-    find "$out" -name \*.so | while read lib; do
-      new_refs="$(patchelf --print-rpath "$lib" | sed -E 's,:?${lib.escape ["+"] openjdk11_headless.outPath}[^:]*,,')"
-      patchelf --set-rpath "$new_refs" "$lib"
-    done
-  '';
+    installPhase = ''
+      cp -r build/modular-sdk $out
+    '';
 
-  disallowedReferences = [ openjdk11_headless ];
+    passthru.deps = deps;
 
-  passthru.deps = deps;
+    meta = with lib; {
+      homepage = "http://openjdk.java.net/projects/openjfx/";
+      license = licenses.gpl2Classpath;
+      description = "The next-generation Java client toolkit; unpacked SDK";
+      maintainers = with maintainers; [ abbradar ];
+      platforms = [ "i686-linux" "x86_64-linux" ];
+    };
+  };
+
+  sdk = common {
+    pname = "openjfx-sdk";
+
+    installPhase = ''
+      cp -r build/sdk $out
+    '';
 
-  meta = with lib; {
-    homepage = "http://openjdk.java.net/projects/openjfx/";
-    license = licenses.gpl2;
-    description = "The next-generation Java client toolkit";
-    maintainers = with maintainers; [ abbradar ];
-    platforms = [ "i686-linux" "x86_64-linux" ];
+    meta = with lib; {
+      homepage = "http://openjdk.java.net/projects/openjfx/";
```

Comment:
```suggestion
      homepage = "https://openjdk.java.net/projects/openjfx/";
```

Reply:
AFAIK https://openjfx.io/ is now used as development space.

Reply:
I'm not sure which URL is appropriate for the `homepage`.

- https://github.com/openjdk/jfx still points to https://openjdk.java.net/projects/openjfx/ from “About”.
- [README.md](https://github.com/openjdk/jfx/blob/5b42b642cefef3129c8925f5496d161ffb0ab285/README.md) says https://openjfx.io/ is the JavaFX community site.
- https://openjdk.java.net/projects/openjfx/ says “OpenJFX Home Page” is https://wiki.openjdk.java.net/display/OpenJFX/Main.
- https://wiki.openjdk.java.net/display/OpenJFX/Main says https://openjfx.io/ is the “OpenJFX Community Site”.